tox.ini

```diff
@@ -71,7 +71,7 @@ commands =
 description = Run charm integration tests
 deps =
     pytest
-    juju==2.9.11 # juju 3.0.0 has issues with retrieving action results
+    juju
```

Comment:
Latest version (3.0.4) seems to work fine, however [this comment](https://github.com/juju/python-libjuju/issues/770#issuecomment-1325351563) mentions that version 3.1.0 would drop support for Juju 2. Should I be pinning to <3.1.0 instead?

Reply:
It seems fine, but I think the right way is to pin to a new python-libjuju 2.9 version (when it's available) that fixes those warnings, so we can get updates and fixes for python-libjuju 2.9, as 2.9 agents will need to be targeted by new versions of python-libjuju 2.9 (not 3.x).

Reply:
I vote to pin 2.9.next when available (after testing) IF will be necessary.
The backward compatibility might stay longer. It will depend on juju 3 stability => popularity.